crates/ruff_python_parser/src/error.rs

```diff
@@ -554,27 +553,60 @@ impl Display for UnsupportedSyntaxError {
                 "Cannot set default type for a type parameter"
             }
         };
+
         write!(
             f,
-            "{kind} on Python {} (syntax was added in Python {})",
+            "{kind} on Python {} (syntax was {changed})",
             self.target_version,
-            self.kind.minimum_version(),
+            changed = self.kind.changed_version(),
         )
     }
 }
 
+/// Represents the kind of change in Python syntax between versions.
+///
+/// Most changes so far have been additions (e.g. `match` and `type` statements), but others are
+/// removals (e.g. parenthesized keyword argument names).
```

Comment:
```suggestion
```

Future readers don't have the context on in which order you added those, so I'm not sure this paragraph is useful to them. 

We could document `Added` and `Removed` but I think it's mostly self explanatory when looking at the `Display` implementation